hardhat/test/unit/01_Storage.t.js

```diff
@@ -0,0 +1,688 @@
+        fixtures.deployStorage
+      );
+      await expect(
+        storage
+          .connect(otherAccount)
+          .addNewTokenAddressByChainId("test", 1, getRandomAddress())
+      ).to.be.reverted;
+    });
+
+    it("should revert if NON admin tries to call updateTokenAddressByChainId", async function () {
+      const { storage, otherAccount } = await loadFixture(
+        fixtures.deployStorage
+      );
+      await expect(
+        storage
+          .connect(otherAccount)
+          .updateTokenAddressByChainId("test", 1, getRandomAddress())
+      ).to.be.reverted;
+    });
+
+    it("should revert if NON admin tries to call batchAddNewTokensAddressesByChainId", async function () {
+      const { storage, otherAccount } = await loadFixture(
+        fixtures.deployStorage
+      );
+      await expect(
+        storage
+          .connect(otherAccount)
+          .batchAddNewTokensAddressesByChainId(
+            ["test"],
+            [1],
+            [getRandomAddress()]
+          )
+      ).to.be.reverted;
+    });
+  });
+
+  describe("Key generator", function () {
+    it("should generate a key from a string", async function () {
+      const { storage } = await loadFixture(fixtures.deployStorage);
+      const key = await storage.getKey("test");
+
+      const ethersKey = ethers.solidityPackedKeccak256(["string"], ["test"]);
+
+      expect(key).to.equal(ethersKey);
+    });
+
+    it("should generate a key from a string & an address", async function () {
+      const { storage } = await loadFixture(fixtures.deployStorage);
+      const preparedAddress = getRandomAddress();
+
+      const key = await storage["getKey(string,address)"](
+        "test",
+        preparedAddress
+      );
+
+      const ethersKey = ethers.solidityPackedKeccak256(
+        ["string", "address"],
+        ["test", preparedAddress]
+      );
+
+      expect(key).to.equal(ethersKey);
+    });
+
+    it("should generate a key from a string & a number", async function () {
+      const { storage } = await loadFixture(fixtures.deployStorage);
+      const preparedNumber = 1;
+
+      const key = await storage["getKey(string,uint256)"](
+        "test",
+        preparedNumber
+      );
+
+      const ethersKey = ethers.solidityPackedKeccak256(
+        ["string", "uint"],
+        ["test", preparedNumber]
+      );
+
+      expect(key).to.equal(ethersKey);
+    });
+  });
+
+  // @todo => add basic setters tests
+  describe("Basic setters", function () {
+    it("should set a new address", async function () {
+      const { storage, owner } = await loadFixture(fixtures.deployStorage);
+      const preparedAddress = getRandomAddress();
+      // const preparedKey = await storage["getKey(string)"]("testKey");
+      const preparedKey = ethers.solidityPackedKeccak256(
+        ["string"],
+        ["testKey"]
+      );
+
+      // THIS CAUSE CONFLICT with ethers 'getAddress()'
+      // const result = await storage.getAddress(preparedKey);
```

Comment:
remove commented code

Reply:
I leave it there until there is no better solution. 
Actual fix is a duplicate of getAddress with getAddr or i delete the comment and create task ?